### PR TITLE
Marvin console improvements

### PR DESCRIPTION
**Pull request by thokkat**
A bit more convenience for marvin console. 

The cursor is now moveable, the last input is preserved while moving through history and the delete key is usable.

### CODE REVIEW

File: game/ui/consolewidget.cpp [hunk 1]
```diff
@@ -137,36 +137,64 @@ void ConsoleWidget::keyDownEvent(KeyEvent& e) {
...
-    if(log.back().size()>0)
-      log.back().pop_back();
+    if(log.back().size()>0 && cursPos>0) {
+      log.back().erase(--cursPos,1);
+      }
+    return;
+    }
+  if(e.key==Event::K_Delete) {
+    if(log.back().size()>cursPos) {
+        log.back().erase(cursPos,1);
```
Comment (Try) [inline game/ui/consolewidget.cpp:191]: nitpick: tabulation

File: game/ui/consolewidget.cpp [hunk 2]
```diff
@@ -137,36 +137,64 @@ void ConsoleWidget::keyDownEvent(KeyEvent& e) {
...
+    cursPos = 0;
+    currCmd = "";
+    histPos = size_t(-1);
     return;
     }
 
   if(e.key==Event::K_Back) {
-    if(log.back().size()>0)
-      log.back().pop_back();
+    if(log.back().size()>0 && cursPos>0) {
```
Comment (Try) [inline game/ui/consolewidget.cpp:184]: Maybe something like 
`if(0<curssPos && curssPos<=log.back().size())`
would be more clean way to express this condition?

**Comment by thokkat** [comment]
Addressed the changes.

**Comment by Try** [comment]
Merged, thanks!

**Pull request merged**